dotnet new streamdeck-plugin -n FirstPlugin build fails

Using the current development branch version of the stream deck plugin template does not build.

Steps to recreate:

1. `dotnet new -i Templates/StreamDeck.PluginTemplate.Csharp`
2. Change directory to a new place like `cd ~/dev/repos`
3. `dotnet new streamdeck-plugin -n FirstPlugin`
4. `cd FirstPlugin`
5. `dotnet build`

Result:

```bash
 dotnet build
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 46.41 ms for /Users/bobby/src/notmyself/FirstPlugin/FirstPlugin.csproj.
MyPlugin.cs(13,4): error CS1997: Since 'MyPlugin.OnKeyUp(StreamDeckEventPayload)' is an async method that returns 'Task', a return keyword must not be followed by an object expression. Did you intend to return 'Task<T>'? [/Users/bobby/src/notmyself/FirstPlugin/FirstPlugin.csproj]
MyPlugin.cs(9,30): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/Users/bobby/src/notmyself/FirstPlugin/FirstPlugin.csproj]
Program.cs(49,29): error CS1501: No overload for method 'Initialize' takes 5 arguments [/Users/bobby/src/notmyself/FirstPlugin/FirstPlugin.csproj]

Build FAILED.

MyPlugin.cs(9,30): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/Users/bobby/src/notmyself/FirstPlugin/FirstPlugin.csproj]
MyPlugin.cs(13,4): error CS1997: Since 'MyPlugin.OnKeyUp(StreamDeckEventPayload)' is an async method that returns 'Task', a return keyword must not be followed by an object expression. Did you intend to return 'Task<T>'? [/Users/bobby/src/notmyself/FirstPlugin/FirstPlugin.csproj]
Program.cs(49,29): error CS1501: No overload for method 'Initialize' takes 5 arguments [/Users/bobby/src/notmyself/FirstPlugin/FirstPlugin.csproj]
    1 Warning(s)
    2 Error(s)

Time Elapsed 00:00:02.76
```

--- a/Templates/StreamDeck.PluginTemplate.Csharp/content/Program.cs
+++ b/Templates/StreamDeck.PluginTemplate.Csharp/content/Program.cs
@@ -1,63 +1,81 @@
 ﻿using McMaster.Extensions.CommandLineUtils;
 using Microsoft.Extensions.Logging;
 using Serilog;
+using Serilog.Settings.Configuration;
 using StreamDeckLib;
 using System;
 using System.IO;
+using System.Reflection;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using System.Diagnostics;
 
 namespace _StreamDeckPlugin_
 {
-	public class Program
-	{
-		static Task<int> Main(string[] args) => CommandLineApplication.ExecuteAsync<Program>(args);
+	    class Program
+    {
+        static ILoggerFactory GetLoggerFactory()
+        {
+            var dir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            Directory.SetCurrentDirectory(dir);
+
+            var configuration = new ConfigurationBuilder()
+                    .AddJsonFile("appsettings.json")
+                    .Build();
+
+            Log.Logger = new LoggerConfiguration()
+                    .ReadFrom.Configuration(configuration)
+                    .CreateLogger();
+
+            var loggerFactory = new LoggerFactory()
+                .AddSerilog(Log.Logger);
+
+            TopLogger = loggerFactory.CreateLogger("top");
+
+            TopLogger.LogInformation("Plugin started");
+
+            return loggerFactory;
+        }
+
+        private static Microsoft.Extensions.Logging.ILogger TopLogger;
+
+        static async Task Main(string[] args)
+        {
 
 
-		[Option(Description = "The port the Elgato StreamDeck software is listening on", ShortName = "port")]
-		public int Port { get; set; }
+#if DEBUG
 
-		[Option(ShortName = "pluginUUID")]
-		public string PluginUUID { get; set; }
+            Debugger.Launch();
 
-		[Option(ShortName = "registerEvent")]
-		public string RegisterEvent { get; set; }
+            while (!Debugger.IsAttached)
+            {
+                await Task.Delay(100);
+            }
 
-		[Option(ShortName = "info")]
-		public string Info { get; set; }
+#endif
+            using (var source = new CancellationTokenSource())
+            {
+                using (var loggerFactory = GetLoggerFactory())
+                {
+                    try
+                    {
 
-		private async Task OnExecuteAsync()
-		{
-			var source = new CancellationTokenSource();
-			var logLocation = Path.Combine(Path.GetTempPath(), $"{GetType().Assembly.GetName().Name}-.log");
+                        await ConnectionManager.Initialize(args, loggerFactory)
+                            .SetPlugin(new $(PluginName)())
+                            .StartAsync(source.Token);
 
-			Log.Logger = new LoggerConfiguration()
-				.MinimumLevel.Verbose()
-				.Enrich
-					.FromLogContext()
-				.WriteTo
-					.File(logLocation, rollingInterval: RollingInterval.Day)
-				.CreateLogger();
+                        Console.ReadLine();
 
-			var loggerFactory = new LoggerFactory()
-				.AddSerilog(Log.Logger);
+                    }
+                    catch (Exception ex)
+                    {
+                        TopLogger.LogError(ex, "Error while running the plugin");
+                    }
+                    source.Cancel();
+                }
 
-			var topLogger = loggerFactory.CreateLogger("top");
-
-			try
-			{
-				await ConnectionManager.Initialize(Port, PluginUUID, RegisterEvent, Info, loggerFactory)
-					.SetPlugin(new $(PluginName)())
-					.StartAsync(source.Token);
-				
-				Console.ReadLine();	
-
-			} catch (Exception ex) {
-				topLogger.LogError(ex, "Error while running the plugin: $(PluginName)");
-			}
-
-			source.Cancel();
-
-		}
-	}
+            }
+        }
+    }
 }

--- a/Templates/StreamDeck.PluginTemplate.Csharp/content/_PluginName_.cs
+++ b/Templates/StreamDeck.PluginTemplate.Csharp/content/_PluginName_.cs
@@ -1,16 +1,52 @@
 ﻿using StreamDeckLib;
 using StreamDeckLib.Messages;
+using System.Dynamic;
 using System.Threading.Tasks;
 
 namespace _StreamDeckPlugin_
 {
 	internal class $(PluginName) : BaseStreamDeckPlugin
 	{
+		private static int _Counter = 0;
+
 		public override async Task OnKeyUp(StreamDeckEventPayload args)
 		{
-			// Your plugin code goes here.
+			_Counter++;
+			await Manager.SetTitleAsync(args.context, _Counter.ToString());
 
-			return Task.CompletedTask;
+			if (_Counter % 10 == 0)
+			{
+				await Manager.ShowAlertAsync(args.context);
+			}
+			else if (_Counter % 15 == 0)
+			{
+				await Manager.OpenUrlAsync(args.context, "https://www.bing.com");
+			}
+			else if (_Counter % 3 == 0)
+			{
+				await Manager.ShowOkAsync(args.context);
+			}
+			else if (_Counter % 7 == 0)
+			{
+				await Manager.SetImageAsync(args.context, "Fritz.png");
+			}
+		}
+
+		public override async Task OnWillAppear(StreamDeckEventPayload args)
+		{
+			if (args.payload != null && args.payload.settings != null && args.payload.settings.counter != null)
+			{
+				_Counter = args.payload.settings.counter;
+			}
+			await Manager.SetTitleAsync(args.context, _Counter.ToString());
+		}
+
+		public override async Task OnWillDisappear(StreamDeckEventPayload args)
+		{
+
+			var settings = new { counter = _Counter };
+
+			await Manager.SetSettingsAsync(args.context, settings);
 		}
 	}
 }

--- a/Templates/StreamDeck.PluginTemplate.Csharp/content/appsettings.json
+++ b/Templates/StreamDeck.PluginTemplate.Csharp/content/appsettings.json
@@ -1,0 +1,16 @@
+﻿{
+  "Serilog": {
+    "MinimumLevel": "Verbose",
+    "WriteTo": [
+      {
+        "Name": "File",
+				"Args": {
+					"path": "log/sampleplugin.log",
+					"rollingInterval": "Day",
+					"outputTemplate": "{Timestamp:yyyy-MM-dd HH:mm:ss.fff} [{Level:u3}] {Message} {NewLine}{Exception}"
+				}
+      }
+    ],
+    "Enrich": [ "FromLogContext" ]
+  }
+}

